csrc/python_frontend/fusion_definition.h

```diff
@@ -71,6 +71,7 @@ struct TORCH_CUDA_CU_API Scalar {
 };
 
 struct TORCH_CUDA_CU_API Vector {
+  Vector() : index(0), size(0), fusion_definition(nullptr) {}
```

Comment:
nitpick: is index 0 noted as an invalid or non-init value, or is it a valid index pointing to something else?

not a big deal since we have fusion_definition==nullptr here. But we might want to assert on fusion_definition != nullptr inside `operator()`

Reply:
I got rid of the default constructor need with some `lambda` usage in the python bindings.  Adding something to the `operator()` did not quite make sense, alone, given that `index` is a public data member of the `struct` and anything code access it.